ios/engine/KMEI/KeymanEngine/Classes/Resource Management/PackageInstallViewController.swift

```diff
@@ -282,20 +284,63 @@ public class PackageInstallViewController<Resource: LanguageResource>: UIViewCon
   }
 
   @objc func installBtnHandler() {
-    // If it is not the root view of a navigationController, just pop it off the stack.
-    if let navVC = self.navigationController, navVC.viewControllers[0] != self {
-      navVC.popViewController(animated: true)
-    } else { // Otherwise, if the root view of a navigation controller, dismiss it outright.  (pop not available)
-      dismiss(animated: true)
-    }
-
     let selectedItems = self.languageTable.indexPathsForSelectedRows ?? []
     let selectedLanguageCodes = selectedItems.map { self.languages[$0.row].id }
 
     let selectedResources = self.package.installableResourceSets.flatMap { $0.filter { selectedLanguageCodes.contains($0.languageID) }} as! [Resource]
 
     self.completionHandler(selectedResources.map { $0.typedFullID })
-    self.associators.forEach { $0.pickerFinalized() }
+
+    let dismissalBlock = {
+      // If it is not the root view of a navigationController, just pop it off the stack.
+      if let navVC = self.navigationController {
+       if navVC.viewControllers[0] != self {
+        navVC.popViewController(animated: true)
+        } else {
+          self.dismiss(animated: true)
+        }
+      } else { // Otherwise, if the root view of a navigation controller, dismiss it outright.  (pop not available)
+        self.dismiss(animated: true)
+      }
+
+      self.associators.forEach { $0.pickerFinalized() }
+    }
+
+    // First, show the package's welcome - if it exists.
+    if let welcomeVC = PackageWebViewController(for: package, page: .welcome) {
+      self.dismissalBlock = dismissalBlock
+
+      let subNavVC = UINavigationController(rootViewController: welcomeVC)
+      _ = subNavVC.view
+
+      let doneItem = UIBarButtonItem(title: NSLocalizedString("command-done",
+                                                              bundle: engineBundle,
+                                                              comment: ""),
+                                     style: .plain,
+                                     target: self,
+                                     action: #selector(self.onWelcomeDismissed))
+
+      // The view's navigation buttoms need to be set on its controller's navigation item,
+      // not the UINavigationController's navigationItem.
+      welcomeVC.navigationItem.rightBarButtonItem = doneItem
+
+      // We need to listen to delegated presentation events on the view-controller being presented.
+      // This version handles iOS 13's "page sheet" slide-dismissal.
+      subNavVC.presentationController?.delegate = self
```

Comment:
The two code lines within this block both took embarrasingly long to get right.  Both times due to the question of "which view controller?" needed to be the base object.